designer/server/src/main/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceImpl.scala

```diff
@@ -646,17 +649,33 @@ class DeploymentServiceImpl(
           params
         )
         customActionOpt = manager.customActionsDefinitions.find(_.actionName == actionName)
-        _ <- existsOrFail(customActionOpt, CustomActionNonExisting(actionName))
+        customAction <- existsOrFail(
+          customActionOpt,
+          CustomActionNonExistingError(
+            s"Couldn't find definition of action ${actionName.value} for scenario ${processIdWithName.name} when trying to validate"
+          )
+        )
+        _ <- validateActionCommand(actionCommand, customAction)
```

Comment:
I can't see any test covering this important case

Reply:
Please specify which line do you mean. This: 
` customAction <- existsOrFail(
          customActionOpt,
          CustomActionNonExistingError(
            s"Couldn't find definition of action ${actionName.value} for scenario ${processIdWithName.name} when trying to validate"
          )
        )
        ` ?

Reply:
No, this:
```
DBIOAction.failed(new IllegalStateException(s"Validation failed for: ${actionCommand.actionName}"))
    }
```

Reply:
Also IMO it shouldn't be 500, it is bad request - not an illegal state of the application

Reply:
I have changed the error type and added a test case.